subsys/bluetooth/mesh/settings.c

```diff
@@ -284,9 +285,9 @@ static struct bt_mesh_rpl *rpl_find(uint16_t src)
 {
 	int i;
 
-	for (i = 0; i < ARRAY_SIZE(bt_mesh.rpl); i++) {
-		if (bt_mesh.rpl[i].src == src) {
-			return &bt_mesh.rpl[i];
+	for (i = 0; i < ARRAY_SIZE(bt_mesh_replay_list); i++) {
+		if (bt_mesh_replay_list[i].src == src) {
+			return &bt_mesh_replay_list[i];
```

Comment:
Seems like this `rpl_find()` and `rpl_alloc()` further below are the reasons why you have to expose the entire struct  outside of the new `rpl.c`? Wouldn't it make sense to move these to `rpl.c` as well and then keep the struct definition and variable internal to `rpl.c`?